core/src/test/java/io/questdb/test/tools/TestUtils.java

```diff
@@ -811,6 +811,68 @@ public static void assertResponse(HttpClient.Request request, int expectedStatus
         }
     }
 
+    /**
+     * Asserts that the actual lines are the same as the expected lines in the reverse order.
+     * For example, assertion with expected: "123\n456\n789\n" and actual: "789\n456\n123\n" passes.
+     * This method expects for the char sequences to end up with a newline character.
+     */
+    public static void assertReverseLinesEqual(@Nullable String message, CharSequence expected, CharSequence actual) {
+        if (expected == null && actual == null) {
+            return;
+        }
+
+        if (expected != null && actual == null) {
+            Assert.fail("Expected: \n`" + expected + "`but have NULL");
+        }
+
+        if (expected == null) {
+            Assert.fail("Expected: NULL but have \n`" + actual + "`\n");
+        }
+
+        if (expected.length() != actual.length()) {
+            Assert.assertEquals(message, reverseLines(expected), actual);
+        }
+
+        if (expected.length() == 0) {
```

Comment:
it would be good to add a comment here why expected would be empty and why arent we asserting actual here